app/src/org/commcare/activities/FullscreenVideoViewActivity.kt

```diff
@@ -0,0 +1,86 @@
+package org.commcare.activities
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import org.commcare.dalvik.databinding.ActivityFullscreenVideoViewBinding
+import org.commcare.views.media.CommCareMediaController
+
+/**
+ * Activity to view inline videos in fullscreen mode, it returns the last time position to the
+ * calling activity
+ *
+ * @author avazirna
+ */
+class FullscreenVideoViewActivity : AppCompatActivity() {
+
+    private lateinit var viewBinding: ActivityFullscreenVideoViewBinding
+    private var lastPosition = -1
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewBinding = ActivityFullscreenVideoViewBinding.inflate(layoutInflater)
+        setContentView(viewBinding.root)
+
+        // Get video URI from intent, finish if no URI is available
+        // TODO: we should inform the user when we finish because there is no data
```

Comment:
instead of crashing, I opted for [canceling the activity and informing the user.](https://github.com/dimagi/commcare-android/pull/2747/commits/123590ee3a6f6c3d3a29fbf89c13c92f9a9544de)